src/createrepo_c.c

```diff
@@ -339,13 +346,15 @@ main(int argc, char **argv)
         exit(EXIT_SUCCESS);
     }
 
-    if (argc != 2) {
-        // No mandatory arguments
-        g_printerr("Must specify exactly one directory to index.\n");
-        g_printerr("Usage: %s [options] <directory_to_index>\n\n",
-                         cr_get_filename(argv[0]));
-        free_options(cmd_options);
-        exit(EXIT_FAILURE);
+    if ( ! cmd_options->split ) {
+        if (argc != 2) {
+            // No mandatory arguments
+            g_printerr("Must specify exactly one directory to index.\n");
+            g_printerr("Usage: %s [options] <directory_to_index>\n\n",
+                     cr_get_filename(argv[0]));
+            free_options(cmd_options);
+            exit(EXIT_FAILURE);
+        }
     }
```

Comment:
Even when ``--split`` is specified, the number of arguments should still be checked, we need handle situation when there is no argument at all.

    $ ./createrepo_c --split
    Directory (null) must exist

is not right